src/main/kotlin/dev/minn/jda/ktx/JDABuilder.kt

```diff
@@ -0,0 +1,399 @@
+            field = value
+        }
+
+    var enableShutdownHook: Boolean = true
+        set(value) {
+            builder.setEnableShutdownHook(value)
+            field = value
+        }
+
+    var autoReconnect: Boolean = true
+        set(value) {
+            builder.setAutoReconnect(value)
+            field = value
+        }
+
+    var eventManager: IEventManager? = null
+        set(value) {
+            builder.setEventManager(value)
+            field = value
+        }
+
+    var audioSendFactory: IAudioSendFactory? = null
+        set(value) {
+            builder.setAudioSendFactory(value)
+            field = value
+        }
+
+    var idle: Boolean = false
+        set(value) {
+            builder.setIdle(value)
+        }
+
+    var activity: Activity? = null
+        set(value) {
+            builder.setActivity(value)
+            field = value
+        }
+
+    var status: OnlineStatus? = null
+        set(value) {
+            if (value != null) {
+                builder.setStatus(value)
+            }
+            field = value
+        }
+
+    var eventListeners: MutableSet<Any> = mutableSetOf()
+
+    var maxReconnectDelay: Int = 900
+        set(value) {
+            builder.setMaxReconnectDelay(value)
+            field = value
+        }
+
+    inline fun sharding(build: InlineSharding.() -> Unit) {
+        val sharding = InlineSharding().also(build)
+        builder.useSharding(sharding.shardId, sharding.shardTotal)
+    }
+
+    var sessionController: SessionController? = null
+        set(value) {
+            builder.setSessionController(value)
+            field = value
+        }
+
+    var voiceDispatchInterceptor: VoiceDispatchInterceptor? = null
+        set(value) {
+            builder.setVoiceDispatchInterceptor(value)
+            field = value
+        }
+
+    var chunkingFilter: ChunkingFilter? = null
+        set(value) {
+            builder.setChunkingFilter(value)
+            field = value
+        }
+
+    var gatewayIntents: MutableSet<GatewayIntent> = mutableSetOf()
+
+    inline fun enableIntents(cacheFlag: GatewayIntent, vararg cacheFlags: GatewayIntent) {
+        builder.enableIntents(cacheFlag, *cacheFlags)
+    }
+
+    inline fun enableIntents(cacheFlags: Set<GatewayIntent>) {
+        builder.enableIntents(cacheFlags)
+    }
+
+    inline fun disableIntents(cacheFlag: GatewayIntent, vararg cacheFlags: GatewayIntent) {
+        builder.disableIntents(cacheFlag, *cacheFlags)
+    }
+
+    inline fun disableIntents(cacheFlags: Set<GatewayIntent>) {
+        builder.disableIntents(cacheFlags)
+    }
+
+    var largeThreshold: Int = 250
+        set(value) {
+            builder.setLargeThreshold(value)
+            field = value
+        }
+
+    var maxBufferSize: Int = 2048
+        set(value) {
+            builder.setMaxBufferSize(value)
+            field = value
+        }
+
+    var injectKtx: Boolean = true
+
+    class InlineSharding {
```

Comment:
This should not be a nested class

Reply:
I was following [`builder.kt`](https://github.com/MinnDevelopment/jda-ktx/blob/master/src/main/kotlin/dev/minn/jda/ktx/builder.kt#L274), as that uses nested classes.